apache/superset
# fix: Visualization settings were lost when editing a datasource from Explore

### SUMMARY

When working on a visualization in Explore, if there were changes to the visualization that had not been "run" via the "Run Query" button, they would be lost when editing the related datasource via the modal on the page.

Why? The reducer that updated the `form_data` when a form component changed was broken in such a way that it would update the control settings to make the change visible, but not the `form_data` settings to make the change durable.

This PR retains the behavior of not updating the encoded data in the URL until the "Run Query" button is pressed.

**Question for the community**: are there any tests for any of the reducers in Superset? Can someone point me to an example? I'd love to write some tests for this change to avoid a regression.

Also, big props to @nytai for walking me through the Explore front-end code and helping me track this one down.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TEST PLAN

* Open the Explore view by creating an arbitrary query in SQL Lab and clicking `Explore`
* Enter some chart options
* Edit the dataset by using the dropdown
* Click `Save` in the modal
* Click `Confirm`
* The chart options you had selected before editing the dataset should still be present.

### ADDITIONAL INFORMATION
